wavesexchange_topic/src/lib.rs

```diff
@@ -608,6 +702,71 @@ mod parse_and_format {
 
             Ok(())
         }
+
+        #[test]
+        fn pairs_one_test() -> anyhow::Result<()> {
+            let topic_data = Topic::parse_str("topic://pairs/amount_asset/price_asset")?.data();
+            let pairs = topic_data
+                .as_pairs()
+                .ok_or(anyhow::anyhow!("bad test case"))?;
+
+            assert_eq!(pairs[0].amount_asset, "amount_asset");
+            assert_eq!(pairs[0].price_asset, "price_asset");
+
+            Ok(())
+        }
+
+        #[test]
+        fn pairs_one_uri_only_test() -> anyhow::Result<()> {
+            let topic_data =
+                Topic::parse_str("topic://pairs/amount_asset/price_asset?pairs[]=skip/skip")?
```

Comment:
This URL should be considered invalid. Parsing code should check such cases and return error.